pkg/cluster/nodes/node.go

```diff
@@ -319,3 +319,59 @@ func (n *Node) WriteKubeConfig(dest string, hostPort int) error {
 
 	return ioutil.WriteFile(dest, buff.Bytes(), 0600)
 }
+
+// WriteFile writes temporary file on the host
+// and copies it to the node
+func (n *Node) WriteFile(dest, content string) error {
+	// create destination directory
+	cmd := n.Command("mkdir", "-p", filepath.Dir(dest))
+	err := exec.RunLoggingOutputOnFail(cmd)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create directory %s", dest)
+	}
+
+	// create temporary file
+	ftmp, err := ioutil.TempFile("", "")
+	if err != nil {
+		return errors.Wrap(err, "failed to create temporary file")
+	}
+
+	path := ftmp.Name()
+	defer os.Remove(path)
+
+	// write content to the temp file
+	_, err = ftmp.WriteString(content)
+	if err != nil {
+		return errors.Wrap(err, "failed to write content to the file")
+	}
+
+	// copy the temp file from the host to the node
+	if err := n.CopyTo(path, dest); err != nil {
```

Comment:
I have a neat trick to avoid this tempfile but we can do it in a follow up:
`node.Command("cp", "/dev/stdin", dest).SetInput(strings.NewReader(contents)).Run()`
Will probably refactor to do this in other places to deal with docker installed via snap not having access to $TMPDIR.

Reply:
looks nice :)